packages/compass-e2e-tests/index.js

```diff
@@ -35,11 +39,36 @@ async function main() {
     mocha.addFile(path.join(__dirname, testPath));
   });
 
-  mocha.run((failures) => (process.exitCode = failures ? 1 : 0));
+  keychain.activate();
+
+  mocha.run((failures) => {
+    keychain.reset();
+    process.exitCode = failures ? 1 : 0;
+  });
 }
 
+process.on('SIGINT', () => {
+  debug(`Process was interrupted. Resetting keychain to default and exiting.`);
+  keychain.reset();
+  process.exitCode = 128 + SIGINT;
+  process.exit();
+});
+
+process.on('SIGTERM', () => {
+  debug(`Process was terminated. Resetting keychain to default and exiting.`);
+  keychain.reset();
+  process.exitCode = 128 + SIGTERM;
+  process.exit();
+});
```

Comment:
Feel free to ignore this, but maybe a little more idiomatic:

```suggestion
process.once('SIGINT', () => {
  debug(`Process was interrupted. Resetting keychain to default and exiting.`);
  keychain.reset();
  process.kill(process.pid, 'SIGINT');
});

process.on('SIGTERM', () => {
  debug(`Process was terminated. Resetting keychain to default and exiting.`);
  keychain.reset();
  process.kill(process.pid, 'SIGTERM');
});
```

Reply:
Just double-check that everything really exits and that everything in posttest runs: https://github.com/mongodb-js/compass/blob/1c146fc2364764876cf689b814e6cd2273adf079/packages/compass-e2e-tests/package.json#L12-L14

I fiddled around with this and the db kept running or something, then ended up leaving it unchanged. But I suppose that's probably because I made it use process.exit(1) - didn't think of process.kill().

Reply:
I was about to ask you how close my code is to what node is doing and if there is a better way. Thanks for suggestion, will change! (it should also be `process.once` for `SIGTERM`, right?)

Reply:
> it should also be `process.once` for `SIGTERM`, right?

Yes :)

> Just double-check that everything really exits and that everything in posttest runs

I don’t think it will run in cases like these either, for any non-zero exit code :/

Reply:
Yeah, postscripts will be skipped if exit code is non-zero unfortunately... We can use mongodb-runner programmaticaly and do a similar clean up for it as this PR introduces for keychain (IIRC it was causing some test stuckage with unfinished processes running before when done like that in other test suites, but maybe it was something else that was causing this)

Reply:
> I don’t think it will run in cases like these either, for any non-zero exit code :/

Aaah so it is basically &&.

Reply:
(this thread got resolved because I merged the suggestion, sorry)

So I moved all the setup/cleanup stuff from the package.json scritps to the index script now (in a pretty rough way, but it seems to work good enough locally) 389fa3723d6796b732e6759f0587b88b5809a056